disorder-core/src/Disorder/Core/Tripping.hs

```diff
@@ -1,10 +1,19 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
 module Disorder.Core.Tripping where
 
 import           Control.Applicative
-
+import           Data.Monoid
 import           Test.QuickCheck
 
 
 tripping :: (Applicative f, Show (f a), Eq (f a)) => (a -> b) -> (b -> f a) -> a -> Property
 tripping to fro a =
   (fro . to) a === pure a
+
+trippingWith :: (Applicative f, Show (f a), Eq (f a), Eq (f Bool)) => (a -> a -> Bool) -> (a -> b) -> (b -> f a) -> a -> Property
```

Comment:
I'm not sure about this, but if we do add it I would make it `(a -> a -> Property)` and `tripping` can use this with `===`